### PR TITLE
=str #18817 make InHandler and OutHandler member classes and add Java API

### PR DESCRIPTION
**Pull request by drewhk**
Alternative to https://github.com/akka/akka/pull/18861 using member types.

### CODE REVIEW

File: akka-stream-tests/src/test/java/akka/stream/stage/JavaIdentityStage.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  private Inlet<T> _in = Inlet.of("Identity.in");
+  private Outlet<T> _out = Outlet.of("Identity.out");
+  private FlowShape<T, T> _shape = FlowShape.of(_in, _out);
+
+  public Inlet<T> in() { return _in; }
+  public Outlet<T> out() { return _out; }
+
+
+  @Override
+  public GraphStageLogic createLogic(Attributes inheritedAttributes) {
```
Comment (drewhk) [inline akka-stream-tests/src/test/java/akka/stream/stage/JavaIdentityStage.java]: Even though it is Java, this stage looks almost as compact as the Scala version.

**Comment by drewhk** [comment]
We can revisit this vesion later

File: akka-stream-tests/src/test/java/akka/stream/stage/JavaIdentityStage.java (hunk 1 above)
Comment (rkuhn) [inline akka-stream-tests/src/test/java/akka/stream/stage/JavaIdentityStage.java]: There is no difference with #18861, though.

Comment (drewhk) [inline akka-stream-tests/src/test/java/akka/stream/stage/JavaIdentityStage.java]: Of course not. The difference is the lack of ThreadLocal, but that is internal.

**Pull request closed without merging**